File: src/m/qmu/expandresponses.py
from helpers import *
from QmuSetupResponses import *


def expandresponses(md, responses):
    
    fnames = fieldnames(responses)

    # maintain order attributes were added
    dresp = OrderedStruct()

    for key in fnames:
        value = getattr(responses, key)
        exec('dresp.{} = type(value)()'.format(key))
        for j in range(len(value)):
            #call setupdesign
            exec('dresp.{}=QmuSetupResponses(md, dresp.{}, value[j])'.format(key, key))

    return dresp

File: src/m/qmu/expandresponses.m
function dresp=expandresponses(md,responses)
%EXPANDRESPONSES - expand a qmu responses structure into a dresp structure array
%
%   Usage:
%      dresp=expandresponses(md,responses)

fnames=fieldnames(responses);

for i=1:length(fnames)

	fhandle=str2func([class(responses.(fnames{i})) '.empty']);
	dresp.(fnames{i})=fhandle();
	for j=1:length(responses.(fnames{i}))
		%call setupdesign
		dresp.(fnames{i})=QmuSetupResponses(md,dresp.(fnames{i}),responses.(fnames{i})(j));
	end
end
